pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+		///
+		/// This is an unsigned call because the caller may not have any fnds to pay fees with.
+		/// This is inspired by Polkadot's claims pallet:
+		/// https://github.com/paritytech/polkadot/blob/master/runtime/common/src/claims.rs
+		///
+		/// This function and the entire concept of unassociated contributions may be obviated if
+		/// They will accept a memo filed in the Polkadot crowdloan pallet.
+		#[pallet::weight(0)]
+		pub fn associate_native_identity(origin: OriginFor<T>, reward_account: T::AccountId, proof: Vec<u8>) -> DispatchResultWithPostInfo {
+			ensure_none(origin)?;
+
+			//TODO check the proof:
+			// 1. Is signed by an actual unassociated contributor
+			// 2. Signs a valid native identity
+
+			//TODO update storage
+
+			//TODO Emit event
+
+			Ok(Default::default())
+		}
+
+		/// Collect whatever portion of your reward are currently vested.
+		#[pallet::weight(0)]
+		pub fn show_me_the_money(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let payee = ensure_signed(origin)?;
+
+			// Calculate the veted amount on demand.
+			let mut info = AccountsPayable::<T>::get(payee).ok_or(Error::<T>::NoAssociatedClaim)?;
+			let now = frame_system::Module::<T>::block_number();
+
+			//TODO This part doesn't compile because of a million stupid errors about converting
+			// between u32, Balance, and BlockNumber. I think that is solvable, just annoying.
+			let payable_per_block = info.total_reward / T::VestingPeriod::get(); //TODO safe math;
+			let payable_period = T::VestingPeriod::get() - info.last_paid;
+			let payable_amount = payable_period * payable_per_block;
+
+			// Update the stored info
+			info.last_paid = now;
+			AccountsPayable::<T>::insert(&payee, &info);
+
+			// Make the payment
+			// TODO where are these reward funds coming from? Currently I'm just minting them right here.
+			// 1. We could have an associated type to absorb the imbalance.
+			// 2. We could have this pallet control a pot of funds, and initialize it at genesis.
+			T::RewardCurrency::deposit_creating(&payee, payable_amount);
+
+			// Emit event
+			Self::deposit_event(Event::RewardsPaid(payee, payable_amount));
+
+			Ok(Default::default())
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// User trying to claim an award did not have an claim associated with it. This may mean
+		/// they did not contribute to the crowdloan, or they have not yet associated a native id
+		/// with their contribution
+		NoAssociatedClaim,
+	}
+
+	#[pallet::storage]
+	pub type AccountsPayable<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, RewardInfo<T>>;
+
+	#[pallet::storage]
+	pub type UnassociatedContributions<T: Config> = StorageMap<_, Blake2_128Concat, T::RelayChainAccountId, RewardInfo<T>>;
+
+	// Design decision:
+	// Genesis config contributions are specified in relay-chain currency
+	// Conversion to reward currency happens when constructing genesis
```

Comment:
We use 18 decimals for our balances and they use 12 decimals on the relay chain right? `xtransfer` uses an associated type to convert by multiplying the relay currency amount by 1_000_000.

https://github.com/PureStake/moonbeam/pull/263/files#diff-0ec06ea58bd455f09ce6b3bb4c2c1c0d37bda51c1e1be2151c560c9c973959ecR536

Reply:
I don't see this conversion happening anywhere. It's likely I don't understand how this pallet works yet in which case ignore it.